src/App.tsx

```diff
@@ -1,24 +1,44 @@
 import React from 'react';
+import { Clock } from './components/Clock';
 import './App.scss';
 
-const App: React.FC = () => {
-  setInterval(() => {
-    const date = new Date();
+type State = {
+  isClockVisible: boolean;
+  clockName: string;
+};
 
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
+class App extends React.Component<{}, State> {
+  state: State = {
+    isClockVisible: true,
+    clockName: 'Default Clock name',
+  };
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
-};
+  changeClockVisibility = () => {
+    this.setState((prevState) => ({ isClockVisible: !prevState.isClockVisible }));
+  };
+
+  changeName = () => {
+    this.setState({ clockName: Math.floor(Math.random() * 100000).toString() });
+  };
+
+  render() {
+    const { isClockVisible, clockName } = this.state;
+
+    return (
+      <div className="App">
+        <h1 className="App__title">React clock</h1>
+        <p className="App__time">
+          Current time:
+          {' '}
+          {isClockVisible && <Clock name={clockName} />}
+        </p>
+        <div className="App__button-set">
+          <button type="button" className="App__button" onClick={this.changeClockVisibility}>{isClockVisible ? 'Hide clock' : 'Show clock'}</button>
+          {isClockVisible && <button type="button" className="App__button" onClick={this.changeName}>Set random name</button>}
```

Comment:
```suggestion
          <button
            type="button"
            className="App__button"
            onClick={this.changeClockVisibility}
          >
            {isClockVisible ? 'Hide clock' : 'Show clock'}
          </button>
          
          {isClockVisible && (
            <button
              type="button"
              className="App__button"
              onClick={this.changeName}
            >
              Set random name
            </button>
          )}
```
It's really hard to read code like this